api/v2/api.go

```diff
@@ -660,6 +671,18 @@ func (api *API) postSilencesHandler(params silence_ops.PostSilencesParams) middl
 	})
 }
 
+// usernameFromHeaderOrBasicAuth returns the username specified as part of Basic Authentication. If no auth is provided,
+// it tries the specified HTTP header of UserHTTPHeader. If none are found it returns an empty string "".
+func (api *API) usernameFromHeaderOrBasicAuth(request *http.Request) string {
+	// First, let's try getting the username from Basic Authentication.
+	if user, _, ok := request.BasicAuth(); ok {
+		return user
```

Comment:
This is not correct, we should have this configured. if a specific header is specified in config, we should always use it, even if a user adds a custom authorization header.

use case for this:

- you put an open id connect proxy in front of alertmanager, which pass the username as http header
- communication between the proxy and alertmanager is done via static basic auth and TLS